core/src/main/scala/dev/profunktor/auth/jwt.scala

```diff
@@ -51,11 +44,10 @@ object jwt {
   ): F[JwtToken] =
     JwtToken(Jwt.encode(jwtClaim, jwtSecretKey.value, jwtAlgorithm)).pure[F]
 
-  def jwtEncode[F[_]: Applicative](
+  def jwtEncode[F[_]: ApplicativeError[*[_], Throwable]](
       jwtClaim: JwtClaim,
-      jwtPublicKey: JwtPublicKey,
-      jwtAlgorithm: JwtAsymmetricAlgorithm
+      jwtPrivateKey: JwtPrivateKey,
   ): F[JwtToken] =
-    JwtToken(Jwt.encode(jwtClaim, jwtPublicKey.value, jwtAlgorithm)).pure[F]
+    Try(JwtToken(Jwt.encode(jwtClaim, jwtPrivateKey.key, jwtPrivateKey.algorithm))).liftTo[F]
```

Comment:
Is `Jwt.encode` impure? If so, prefer to use `F.catchNonFatal` here as well.

Reply:
Java methods are used there throws some exceptions

done